### PR TITLE
Test automation refactor

### PR DESCRIPTION
**Pull request by mwang2016**

### CODE REVIEW

File: test/lib/commonkafka.py [hunk 1]
```diff
@@ -108,80 +96,100 @@ def get_kafka_connector_tasks(setup, params):
...
+    '''
+    time.sleep(2)
+    response = requests.get(url=setup["kafka_connect_url"] + "/connectors/" + params["name"] + "/status",
+                            headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
+    content = response.json()
 
-    return None
+    if content.get('connector'):
+        if content['connector']['state'] == 'RUNNING':
+            task_status = jsonpath.jsonpath(content, '$.tasks.*.state')
```
Comment (rockb1017) [inline test/lib/commonkafka.py:127]: if there are multiple tasks, what does it return ?

File: test/lib/data_gen.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "name": "",
+            "connector_class": "com.splunk.kafka.connect.SplunkSinkConnector",
+            "tasks_max": "1",
+            "topics": config["kafka_topic"],
+            "splunk_indexes": config["splunk_index"],
+            "splunk_hec_uri": config["splunk_hec_url"],
+            "splunk_hec_token": config["splunk_token"],
+            "splunk_hec_raw": "false",
+            "splunk_hec_ack_enabled": "false",
+            "splunk_hec_ssl_validate_certs": "false",
```
Comment (rockb1017) [inline test/lib/data_gen.py:24]: shouldn't we use `.` instead of `_`?

File: test/testcases/test_crud.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+        # Validate restart task
+        assert restart_kafka_connector(setup, connector_definition) == expected
+
+        # Validate delete task
+        assert delete_kafka_connector(setup, connector_definition) == expected
+
+    @pytest.mark.parametrize("test_case, config_input, expected", [
+        ("test_invalid_tasks_max", {"name": "test_invalid_tasks_max", "tasks_max": "dummy-string"}, False),
+        ("test_invalid_splunk_hec_raw", {"name": "test_invalid_splunk_hec_raw", "splunk_hec_raw": "disable"}, False),
```
Comment (rockb1017) [inline test/testcases/test_crud.py:89]: here as well, `.`

File: test/lib/commonkafka.py (hunk 1 above)
Comment (mwang2016) [reply test/lib/commonkafka.py:127]: Yes, if I set `"tasks.max": "3"` there will be 3 tasks. I also have test cases to cover that to make sure all the tasks state are 'FAILED'

File: test/lib/data_gen.py (hunk 2 above)
Comment (mwang2016) [reply test/lib/data_gen.py:24]: These are the keys for data_gen so we can update the payload easily through the template. In the connector.template, they are all `.`

Comment (rockb1017) [reply test/lib/data_gen.py:24]: approved. personally i think it is better to be identical. to reduce any possible confusion

**Pull request merged**